web-common/src/features/explores/selectors.ts

```diff
@@ -131,123 +124,32 @@ export async function fetchExploreSpec(
   }
 
   const defaultExplorePreset = getDefaultExplorePreset(
-    exploreResource.explore.state?.validSpec ?? {},
+    exploreSpec,
     fullTimeRange,
   );
+  const { partialExploreState: initExploreState, errors } =
+    convertPresetToExploreState(
+      metricsViewSpec,
+      exploreSpec,
+      defaultExplorePreset,
+    );
+
+  let initLoadedOutsideOfURL = false;
+  const stateFromLocalStorage = restorePersistedDashboardState(
+    exploreSpec,
+    (prefix ?? "") + exploreName,
+  );
+  if (stateFromLocalStorage) {
+    initLoadedOutsideOfURL = true;
+    Object.assign(initExploreState, stateFromLocalStorage);
+  }
```

Comment:
We should not be doing this assignment/synthesis here. We should do it in one central place with the rest of the state synthesis. I think that place is `DashboardURLStateSync.svelte`.